src/integrationTest/java/org/openstreetmap/atlas/generator/slicing/DynamicRelationSlicingTest.java

```diff
@@ -0,0 +1,118 @@
+    {
+        ResourceFileSystem.addResource(path, new InputStreamResource(
+                () -> DynamicRelationSlicingTest.class.getResourceAsStream(name)));
+    }
+
+    /**
+     * This integration test country slices a water relation that spans four shards, and crosses a
+     * country boundary. The relation is country sliced twice, once from the ABC perspective and
+     * once from DEF. The resulting closed relations then have their areas validated.
+     */
+    @Test
+    public void testRelationSlicing()
+    {
+        // parameter setup
+        final Sharding sharding = new DynamicTileSharding(new File(
+                DynamicRelationSlicingTest.class.getResource("simpleSharding.txt").getFile()));
+
+        final CountryBoundaryMap boundaryMap = CountryBoundaryMap
+                .fromPlainText(new InputStreamResource(() -> DynamicRelationSlicingTest.class
+                        .getResourceAsStream("simpleBoundaryFile.txt")));
+
+        final Map<String, String> sparkOptions = new HashMap<>();
+        sparkOptions.put("spark.executor.memory", "512m");
+        sparkOptions.put("spark.executor.memory", "512m");
+        sparkOptions.put("spark.driver.memory", "512m");
+        ResourceFileSystem.simpleconfiguration()
+                .forEach((key, value) -> sparkOptions.put(key, value));
+
+        final HadoopAtlasFileCache lineSlicedAtlasCache = new HadoopAtlasFileCache(
+                "resource://test/atlas/", sparkOptions);
+
+        final CountryShard initialShardABC = CountryShard.forName("ABC_1-0-0");
+        final CountryShard initialShardDEF = CountryShard.forName("DEF_1-1-1");
+
+        final Function<Shard, Optional<Atlas>> atlasFetcherABC = AtlasGeneratorHelper.atlasFetcher(
+                lineSlicedAtlasCache, lineSlicedAtlasCache, boundaryMap, "ABC",
+                initialShardDEF.getShard());
+        final Function<Shard, Optional<Atlas>> atlasFetcherDEF = AtlasGeneratorHelper.atlasFetcher(
+                lineSlicedAtlasCache, lineSlicedAtlasCache, boundaryMap, "DEF",
+                initialShardDEF.getShard());
+
+        // the operation we are testing
+        final Atlas relationSlicedAtlasDEF = new RawAtlasCountrySlicer("DEF", boundaryMap, sharding,
+                atlasFetcherDEF).sliceRelations(initialShardDEF.getShard());
+        final Atlas relationSlicedAtlasABC = new RawAtlasCountrySlicer("ABC", boundaryMap, sharding,
+                atlasFetcherABC).sliceRelations(initialShardABC.getShard());
+
+        // Sum the areas and check it is the size expected
+        double totalAreaDEF = 0;
+        for (final Line line : relationSlicedAtlasDEF.lines())
+        {
+            if (line.isClosed())
+            {
+                final Polygon polygon = new Polygon(line.asPolyLine());
+                totalAreaDEF += polygon.surface().asDm7Squared();
+            }
+        }
+        Assert.assertTrue(totalAreaDEF == 5.9870426059188685E17);
```

Comment:
Maybe use assertEquals here so if it diverges in the future it will tell what it saw and that it is different?